src/hotspot/share/cds/dumpTimeClassInfo.cpp

```diff
@@ -74,6 +74,10 @@ DumpTimeClassInfo::DumpTimeClassInfo(const DumpTimeClassInfo& src) {
       }
     }
   }
+
+  // The saved cp entries array is not modified by the ArchiveBuilder, so there's
+  // no need to make a deep copy. It will be freed when the holder class is removed.
+  _saved_cpcache_entries = src._saved_cpcache_entries;
```

Comment:
If you're copying these entries, should you make src._saved_cpcache_entries null so you don't double delete them?  Or just for safety in case someone adds deleting it to the destructor later.

Reply:
Actually, the `_saved_cpcache_entries` has a different lifecycle than the `DumpTimeClassInfo`. We set the `_saved_cpcache_entries` after the cpCache is allocated, and it's never changed afterwards. It's freed when the cpCache is deallocated.

I can see why the current code is confusing. I'll try to move it to a new table of `cpCache*` -> `ConstantPoolEntry*` which will make the code easier to understand and maintain.